rust/lance-arrow/src/lib.rs

```diff
@@ -783,7 +783,8 @@ fn project(struct_array: &StructArray, fields: &Fields) -> Result<StructArray> {
             )));
         }
     }
-    StructArray::try_new(fields.clone(), columns, None)
+    // Preserve the struct's validity when projecting
+    StructArray::try_new(fields.clone(), columns, struct_array.nulls().cloned())
```

Comment:
Nice catch.  Guess we haven't had to worry about struct nullability in the table for a while :smile: